src/it/unibs/ing/ingsw/category/Category.java

```diff
@@ -16,38 +16,57 @@
      * @param description Descrizione della categoria
      * @param isRootCategory 'true' se la categoria è radice, 'false' altrimenti
      */
-    public Category(String name, String description, boolean isRootCategory) {
+    public Category(String name, String description, boolean isRootCategory, List<Field> newFields) {
+        if (newFields == null){
+            newFields = new ArrayList<>();
+        }
+
         this.name = name;
         this.description = description;
 
         fields = new ArrayList<>();
+
         if (isRootCategory) {
             fields.add(new Field<String>(true, "Stato di conservazione"));
             fields.add(new Field<String>(false, "Descrizione libera"));
         }
+        fields.addAll(newFields);
 
         children = new HashMap<>();
         parent = null;
     }
 
+    public Category(String name, String description, List<Field> newFields) {
+        this(name, description, false, newFields);
+    }
+
     /**
      * Costruttore Category figlia
      * @param name Nome della categoria
      * @param description Descrizione della categoria
      */
     public Category(String name, String description) {
-        this(name, description, false);
+        this(name, description, false, null);
     }
 
     /**
      * Aggiunge una categoria figlia
      * @param childCategory La categoria figlia da aggiungere
      */
     public void addChildCategory(Category childCategory) {
+        addAllFieldsToCategory(childCategory);
```

Comment:
Se facciamo le copie da genitore a figlio allora parent non ci serve più (o sbaglio?)

Reply:
Parent non ci serve perché per il modo in cui creiamo i figli sappiamo chi é il padre e già a costruttore scatta l'ereditarietà. Questo possiamo farlo solo perché le categorie non si possono modificare a runtime, altrimenti dovremmo tenere traccia anche dei parents e delle modifiche ai parents

Reply:
Io lo farei con `parent` però, che anche se non è richiesto il codice diventa più riusabile e robusto.

Ma soprattutto credo sia anche più facile da fare (dovrebbe bastare una funzione ricorsiva). Se non è banale invece lasciamo così.